### PR TITLE
[dvc][admin-tool][controller][router] Cleanup LeaderFollowerModel related checks

### PR DESCRIPTION
**Pull request by sushantmane**
## Cleanup LeaderFollowerModel-related checks

Resolves #XXX

## How was this PR tested?
InternalCI  + GHCI

## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

**Comment by sixpluszero** [comment]
But isn't L/F the only base mode today? Why setLeaderFollowerEnabled(true) is still needed?

**Comment by sushantmane** [comment]
> But isn't L/F the only base mode today? Why setLeaderFollowerEnabled(true) is still needed?

To make sure that some other components which do not yet have this patch yet do get `isLeaderFollowerEnabled == true` in the response. And some logic is triggered based on this check.

File: internal/venice-common/src/main/java/com/linkedin/venice/meta/Store.java [hunk 1]
```diff
@@ -248,9 +248,12 @@ static boolean isSystemStore(String storeName) {
...
   void setBootstrapToOnlineTimeoutInHours(int bootstrapToOnlineTimeoutInHours);
 
-  boolean isLeaderFollowerModelEnabled();
+  default boolean isLeaderFollowerModelEnabled() {
+    return true;
+  }
 
-  void setLeaderFollowerModelEnabled(boolean leaderFollowerModelEnabled);
+  default void setLeaderFollowerModelEnabled(boolean leaderFollowerModelEnabled) {
+  }
```
Comment (nisargthakkar) [inline internal/venice-common/src/main/java/com/linkedin/venice/meta/Store.java:256]: Can you add a `@deprecated` annotation to both of these?

File: internal/venice-common/src/main/java/com/linkedin/venice/meta/Version.java [hunk 2]
```diff
@@ -98,11 +98,14 @@ public static PushType valueOf(int value) {
...
-  boolean isLeaderFollowerModelEnabled();
+  default boolean isLeaderFollowerModelEnabled() {
+    return true;
+  }
 
   boolean isNativeReplicationEnabled();
 
-  void setLeaderFollowerModelEnabled(boolean leaderFollowerModelEnabled);
+  default void setLeaderFollowerModelEnabled(boolean leaderFollowerModelEnabled) {
+  }
```
Comment (nisargthakkar) [inline internal/venice-common/src/main/java/com/linkedin/venice/meta/Version.java:108]: Same about the `@deprecated` annotation to the LF checks here

File: services/venice-controller/src/main/java/com/linkedin/venice/controller/ZkHelixAdminClient.java [hunk 3]
```diff
@@ -246,14 +246,13 @@ public void createVeniceStorageClusterResources(
...
       int numberOfPartition,
-      int replicationFactor,
-      boolean isLeaderFollowerStateModel) {
+      int replicationFactor) {
     if (!helixAdmin.getResourcesInCluster(clusterName).contains(kafkaTopic)) {
       helixAdmin.addResource(
           clusterName,
           kafkaTopic,
           numberOfPartition,
-          isLeaderFollowerStateModel ? LeaderStandbySMD.name : VeniceStateModel.PARTITION_ONLINE_OFFLINE_STATE_MODEL,
```
Comment (nisargthakkar) [inline services/venice-controller/src/main/java/com/linkedin/venice/controller/ZkHelixAdminClient.java:256]: Maybe this doesn't come in scope of this PR, but we should remove `VeniceStateModel.PARTITION_ONLINE_OFFLINE_STATE_MODEL` too

**Review by nisargthakkar — COMMENTED** [review]
Looks great. Just some minor nits

File: internal/venice-common/src/main/java/com/linkedin/venice/meta/Store.java (hunk 1 above)
Comment (sushantmane) [reply internal/venice-common/src/main/java/com/linkedin/venice/meta/Store.java:256]: Done

File: internal/venice-common/src/main/java/com/linkedin/venice/meta/Version.java (hunk 2 above)
Comment (sushantmane) [reply internal/venice-common/src/main/java/com/linkedin/venice/meta/Version.java:108]: Done

**Comment by sushantmane** [comment]
Thanks for the quick review, Nisarg! I've addressed the comments.

**Review by nisargthakkar — APPROVED** [review]
Thank you, Sushant. Great job. Keep the red PRs coming

https://www.youtube.com/watch?v=Vwgdc_Tqgvs

**Comment by sushantmane** [comment]
> Thank you, Sushant. Great job. Keep the red PRs coming
> 
> https://www.youtube.com/watch?v=Vwgdc_Tqgvs

slowly but surely!

**Pull request merged**